cmd/dm-master/main_test.go

```diff
@@ -34,19 +41,20 @@ func TestRunMain(t *testing.T) {
 		}
 	}
 
-	waitCh := make(chan int, 1)
 	os.Args = args
+	wg.Add(1)
 	go func() {
-		main()
+		defer wg.Done()
+		mainWrapper(exitCh)
 		close(waitCh)
 	}()
 
 	signalCh := make(chan os.Signal, 1)
 	signal.Notify(signalCh, syscall.SIGINT, syscall.SIGQUIT, syscall.SIGTERM, syscall.SIGHUP)
 	select {
-	case <-signalCh:
-		return
 	case <-waitCh:
-		return
+	case <-signalCh:
+		close(exitCh)
```

Comment:
our purpose is to control
1. exit the process when received signals
2. wait for the process exit
?

seems we call `signal.Notify` both in `main.go` and `main_test.go`. Is it needed?

Reply:
well the test starts `main` in a single goroutine, so it catches singal in `main_test.go` too, which is not necessary. I simplify the code by call `main()` directly